appengine-deploy/src/main.ts

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright 2020 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import * as core from '@actions/core';
+import * as exec from '@actions/exec';
+import * as setupGcloud from '../../pkg/dist/index';
+
+async function run(): Promise<void> {
+  try {
+    // Get action inputs.
+    const projectId = core.getInput('project_id', { required: true });
```

Comment:
Does this need to be required? If the user set the project via gcloud, then this wouldn't be required.

Reply:
This is for the case that gcloud is authenticated with one project/service account but wanting to deploy to another project (with the correct permissions).

Reply:
Right, but I don't think it should be _required_. It could be optional right? I think we should remove the `required: true` attribute here.

Reply:
Got it, I see what you mean now.